src/Doctrine/Functions/Rand.php

```diff
@@ -33,7 +33,7 @@ public function parse(\Doctrine\ORM\Query\Parser $parser): void
         $parser->match(Lexer::T_IDENTIFIER);
         $parser->match(Lexer::T_OPEN_PARENTHESIS);
 
-        if ($lexer->lookahead['type'] !== Lexer::T_CLOSE_PARENTHESIS) {
+        if ($lexer->lookahead->type !== Lexer::T_CLOSE_PARENTHESIS) {
```

Comment:
This is because in the CI it migrated silently to lexer v3? What if another dependency is still requiring v2? It was an array before in v2, so we need to have something to guard against that.

Reply:
The dependency removal has unlocked some dependency upgrades like Lexer and Doctrine.

All libraries who need Lexer to accept the v2 or v3:
```
$ symfony composer why doctrine/lexer
doctrine/annotations    2.0.2   requires  doctrine/lexer (^2 || ^3)     
doctrine/orm            2.20.2  requires  doctrine/lexer (^2 || ^3)     
egulias/email-validator 4.0.3   requires  doctrine/lexer (^2.0 || ^3.0) 
symfony/doctrine-bridge v5.4.48 conflicts doctrine/lexer (<1.1)         
symfony/validator       v5.4.48 conflicts doctrine/lexer (<1.1)   
```

Reply:
Let's go for `doctrine/lexer: ^2 || ^3`, as we use the lexer internally in code we need to mark is as own dependency anyway. Looks like 2 added object access and contained both array and object access options, so the code should be fine.

Reply:
Dependency added. Ready to merge.